proposals/spline-animation/README.md

```diff
@@ -0,0 +1,1980 @@
+
+Height and length are both specified multiplied by 3; e.g. if a tangent vector
+is 1.5 units in the time dimension, its length is recorded as 4.5.
+
+Heights are positive for upward-sloping post-tangents, and negative for
+upward-sloping pre-tangents.
+
+### Continuity
+
+Segment-to-segment continuity, as in all Beziers, depends on the alignment of a
+knot's pre-tangent and post-tangent.  Continuity is not explicitly recorded as
+part of spline data, and is not automatically preserved by USD editing
+operations, but it may be ensured programmatically by clients, and it may be
+queried from a spline.  These are the cases:
+
+| Continuity class | Continuous value | Continuous tangents | Continuous slope | How achieved                          |
+| :--------------- | :--------------: | :-----------------: | :--------------: | :------------------------------------ |
+| Discontinuous    | No               | No                  | No               | Dual-valued knot                      |
+| C0               | Yes              | No                  | No               | Broken tangents (mismatched slopes)   |
+| G1               | Yes              | Yes                 | No               | Unequal tangents (mismatched lengths) |
+| C1               | Yes              | Yes                 | Yes              | Identical tangents                    |
+
+![Discontinuous](./discontinuous.png)
+![C0](./C0.png)
+![G1](./G1.png)
+![C1](./C1.png)
+
+The higher continuities G2 and C2 are sometimes useful, but for now we are
+proposing omitting them from the queries supported by USD splines.  That is for
+two reasons:
+
+- For G2 and C2, there aren't intuitive rules about how tangents align.
+  Instead, more abstract geometric invariants must be maintained.
+
+- Achieving G2 and C2 requires relinquishing some degree of "local control",
+  which is typically a vital benefit of Bezier splines.  Local control means
+  that a knot or tangent position may be changed without affecting the curve on
+  any segments other than its own.
+
+### Automatic Tangents
+
+Sometimes spline authors want to specify only knot values, and get a "nice"
+curve that interpolates those knots smoothly.  This sacrifices some degree of
+control over the shape of the curve, but allows quicker work, and guarantees
+consistently styled shapes with some degree of continuity.
+
+USD Anim will allow any tangent to be specified as "auto".  Each of the two
+tangents of a knot may be specified separately, so a knot may be "pre-auto" and
+"post-manual", or vice versa.
+
+Automatic tangents are computed only when knots are first defined, and when they
+are edited.  We therefore say that automatic tangents are an _edit-time
+behavior_.
+
+The exact algorithm for computing automatic tangents has yet to be specified;
+most likely we will adopt one of the Maya auto-tangent algorithms.  Other
+algorithms are possible, and may be available in the future.  Ideally any
```

Comment:
Totally understand the desire, but we're trying to avoid extensions, so that all content can be handled by all clients.  If there's no need for other clients to edit your knots, would it suffice to run your own auto-tangent algorithm on top of USD splines, without the auto-tangent behavior living inside the USD core?

Another avenue would be that if you think your auto-tangent algorithm is awesome and universally useful and you're willing to give it away, you could propose contributing it to USD.